### PR TITLE
Map user to groups

### PR DESCRIPTION
**Pull request by bbockelm**
(Resubmitting PR for main repo)

If CVMFS is present on the TW or the schedd, set the CMS group names for the user's jobs.

With this patch, if @juztas submits a task, he should see the following attribute added to the DAG job:

CMSGroups = "T2_US_Nebraska"

- James is making the corresponding changes to the pilot environment to obey the CMSGroups attribute in ITB.
- Farrukh is adding the group to ITB for launching the local pilots.

@jamesletts

### CODE REVIEW

File: src/python/TaskWorker/Actions/PreJob.py [hunk 1]
```diff
@@ -298,6 +301,16 @@ def alter_submit(self, crab_retry):
         ## Add the site black- and whitelists and the DESIRED_SITES to the
         ## Job.<job_id>.submit content.
         new_submit_text = self.redo_sites(new_submit_text, crab_retry, use_resubmit_info)
+
+        ## Add group information:
+        username = self.task_ad.get('CRAB_UserHN')
+        if 'CMSGroups' in self.task_ad:
+            new_submit_text += '+CMSGroups = %s\n' % classad.quote(groups)
```
Comment (mmascher) [inline src/python/TaskWorker/Actions/PreJob.py]: I think groups is undefined. Did you miss a `groups = self.task_ad.get('CMSGroups')`?

I will try to run the code after dinner before merging it.

**Comment by bbockelm** [comment]
Good catch!  It should have referred to the task_ad.  Fixed.

**Comment by bbockelm** [comment]
I have added 'mmascher' to the local-users.txt for T2_US_Nebraska.  When you test out this patch, it should have:

CMSGroups="T2_US_Nebraska"

in generated task ad.  IIRC, it will take 30-60 minutes for this to propagate into CVMFS (wait until your HN name is in /cvmfs/cms.cern.ch/SITECONF/T2_US_Nebraska/GlideinConfig/local-users.txt before testing).

**Comment by mmascher** [comment]
I will let Justas test this, my username is not there yet. Anyhow, I tested the code to make sure it was not breaking anything, and I included it on 3.3.16.rc2.

**Comment by bbockelm** [comment]
Weird - I will bug Bockjoo about why CVMFS is not updating.

**Pull request merged**